modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalSpec.java

```diff
@@ -111,6 +136,13 @@ public void setSpecifications(Collection<RatedRequest> specifications) {
                 throw new ParsingException(p.getTokenLocation(), "error parsing rank request", ex);
             }
         } , METRIC_FIELD);
+        PARSER.declareObject(RankEvalSpec::setScript, (p, c) -> {
+            try {
+                return Script.parse(p, c.getParseFieldMatcher(), "mustache");
+            } catch (IOException ex) {
+                throw new ParsingException(p.getTokenLocation(), "error parsing rank request", ex);
+            }
+        }, SCRIPT_FIELD);
```

Comment:
If you are referring to https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/script/Script.java#L194 ... this needs a QueryParseContext as parameter, all I have is a RankEvalParseContext.

After chatting on Slack we decided to change the signature of the method to take a ParseFieldMatcher and the ScriptLanguage so that fits for this use case as well.

Turns out this was the original parsing method's signature. So simply removed the method @cbuescher added, inserted a try/catch in the original parsing method instead.